### ISSUE TITLE
Memetic: Configurable quantum step size based on training progress

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1330]
## Parent Issue
This is a sub-issue of #1321 (Plan to improve memetic evolution)

## Summary
The current MIN_STEP quantum size is a fixed constant (0.000_000_1). Adaptive step sizing based on training progress could improve convergence - larger steps when far from optimum, smaller steps when fine-tuning.

## Current Implementation
In `src/blackbox/FineTune.ts:11`:
```typescript
export const MIN_STEP = 0.000_000_1;
```

This is used throughout fine-tuning but never adapts.

## Proposed Improvement
- Track training progress and error magnitude
- Use larger quantum steps early in training or when errors are large
- Use smaller quantum steps when close to convergence
- Implement a step size schedule or adaptive mechanism
- Add configuration option to control step size behaviour

## Technical Approach
1. Calculate error/score improvement rate over recent generations
2. Implement step size as: `baseStep * (1 + errorScale * normalizedError)`
3. Add bounds: minimum step (current MIN_STEP) and maximum step
4. Make step size configurable in NeatOptions

## Expected Benefits
- Faster exploration when far from optimum
- Finer adjustments near convergence
- Better balance between exploration and exploitation

## Requirements
- [ ] Follow TDD - write failing tests first
- [ ] Follow DRY principles - avoid code duplication
- [ ] Unit tests must verify correct behaviour (not performance)
- [ ] If performance improvement is claimed, include benchmark results as evidence
- [ ] Use Australian English spelling (e.g., behaviour, optimise)

**Comment by stservice** [comment]
## Automated Processing Failed (First Attempt)

The automated worker was unable to complete this issue on the first attempt.

### Failure Details
Git push failed while pushing branch `issue-1330-memetic-configurable-quantum-step-size-based-on-tr` to origin.

This typically happens due to:
- Authentication/permission issues
- Protected branch rules
- Network connectivity problems
- Missing `workflow` scope for OAuth tokens (when modifying GitHub Actions files)

<details>
<summary>Git push error output (click to expand)</summary>

```
To https://github.com/stSoftwareAU/NEAT-AI.git
 ! [rejected]          issue-1330-memetic-configurable-quantum-step-size-based-on-tr -> issue-1330-memetic-configurable-quantum-step-size-based-on-tr (fetch first)
error: failed to push some refs to 'https://github.com/stSoftwareAU/NEAT-AI.git'
hint: Updates were rejected because the remote contains work that you do not
hint: have locally. This is usually caused by another repository pushing to
hint: the same ref. If you want to integrate the remote changes, use
hint: 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```
</details>

### What happens next?
- This issue has been marked with the `failed-once` label
- The worker will **automatically retry** this issue on the next scan
- If it fails again, the issue will be marked as `failed` and require manual intervention

### To prevent automatic retry
If you want to work on this manually, remove the `failed-once` label and assign yourself.

**Comment by stservice** [comment]
## Automated Processing Failed (Second Attempt - Permanently Failed)

The automated worker has failed to complete this issue twice and will not retry automatically.

### Failure Details
Git push failed while pushing branch `issue-1330-memetic-configurable-quantum-step-size-based-on-tr` to origin.

This typically happens due to:
- Authentication/permission issues
- Protected branch rules
- Network connectivity problems
- Missing `workflow` scope for OAuth tokens (when modifying GitHub Actions files)

<details>
<summary>Git push error output (click to expand)</summary>

```
To https://github.com/stSoftwareAU/NEAT-AI.git
 ! [rejected]          issue-1330-memetic-configurable-quantum-step-size-based-on-tr -> issue-1330-memetic-configurable-quantum-step-size-based-on-tr (fetch first)
error: failed to push some refs to 'https://github.com/stSoftwareAU/NEAT-AI.git'
hint: Updates were rejected because the remote contains work that you do not
hint: have locally. This is usually caused by another repository pushing to
hint: the same ref. If you want to integrate the remote changes, use
hint: 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```
</details>

### What happened?
- This issue failed on the first attempt and was marked with `failed-once`
- The worker retried and failed again on the second attempt
- The issue has been marked with the `failed` label to prevent repeated attempts
- Manual intervention is required

### Why this might be happening
- The issue may be too complex or large for automated processing
- The issue description may need more detail or clarification
- There may be dependencies or permissions that Claude doesn't have

### To retry this issue
1. Investigate and address the underlying problem (check the failure details above)
2. Consider simplifying or breaking down the issue into smaller tasks
3. Remove the `failed` label from this issue
4. The worker will pick it up again on the next scan

**Comment by stservice** [comment]
## Automated Processing Failed (First Attempt)

The automated worker was unable to complete this issue on the first attempt.

### Failure Details
Claude timed out after 3600 seconds (60 minutes).

This issue may be too complex, too large, or require additional permissions that Claude doesn't have.

<details>
<summary>Last output from Claude (click to expand)</summary>

```
No output captured
```
</details>

### What happens next?
- This issue has been marked with the `failed-once` label
- The worker will **automatically retry** this issue on the next scan
- If it fails again, the issue will be marked as `failed` and require manual intervention

### To prevent automatic retry
If you want to work on this manually, remove the `failed-once` label and assign yourself.
